### PR TITLE
FEATURE : Create an item

### PR DESCRIPTION
**Pull request by louisVHZ**
## Content

Ajout de la fonctionnalité "Create an item" avec ses tests.

---

## Checklist

Please check that your pull request is correct:

- Each commit:
    - [x] corresponds to a contribution that should be notified to users,
    - [x] does not generate new errors or warnings at compile or test time,
    - [x] must be attributed to its real authors (with correct GitHub IDs and [correct syntax for multiple authors](https://help.github.com/articles/creating-a-commit-with-multiple-authors/)).
- The title of a commit should:
    - [x] begin with a contribution type
        - `FEATURE` for a behaviour allowing a user to do something new,
        - `FIX` for a behaviour which has been changed in order to meet user’s expectations,
        - `TEST` when it concerns an acceptance test,
        - `PROCESS` for a change in the way the software is built, tested, deployed,
        - `DOC` when it concerns only internal documentation (however it is better to combine it with the contribution that required this documentation change),
    - [x] be followed by a colon (`:`) with one space after and no space before,
    - [x] be followed by a title (written in English) as short, as user-centered and as explicit as possible
        - If it is a feature, the title must be the user action (beginning with a verb, and please not `manage`),
        - If it is a fix, the title must describe the intended behavior (with `should`).
    - [x] ends with a reference to the corresponding ticket with the following syntax:
        - `(closes #xx)` if xx is a feature ticket (and the commit is a complete implementation),
        - `(fixes #xx)` if xx is a fix ticket (and the commit is a complete fix),
        - `(see #xx)` otherwise,
- Each committed line is:
    - [x] useful (it would not work if removed)
        - if it is a comment line, its information could not be conveyed by better variables and function naming, better code structuring, or better commit message,
    - [x] related to this very contribution (feature, fix...),
    - [x] in English (with the exception of Gherkin scenarios in French and resulting steps),
    - [x] without any typo in variable, class or function names,
    - [x] correctly indented (spaces rather than tabs, same number of characters as in the rest of the file).

### CODE REVIEW

File: features/step_definitions/item.rb [hunk 1]
```diff
@@ -1,26 +1,39 @@
 require 'capybara/cucumber'
-require 'capybara/cuprite'
+require 'selenium/webdriver'
```
Comment (benel) [inline features/step_definitions/item.rb]: Please don't change the development environment. It makes tests on the continuous integration server to fail.

File: features/step_definitions/item.rb [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
 
 Capybara.run_server = false
-Capybara.default_driver = :cuprite
-Capybara.javascript_driver = :cuprite
+Capybara.default_driver = :selenium_chrome_headless
```
Comment (benel) [inline features/step_definitions/item.rb]: Idem.

**Review by benel — CHANGES_REQUESTED** [review]
The tests fail on Travis.
Please note that you updated tools settings to conform your local configuration... Therefore it cannot work on the continuous integration server.
Please try to base your 2 commits on v7 HEAD (and remove your local hacks).
Hopefully it will work.

File: features/step_definitions/portfolio.rb [hunk 3]
```diff
@@ -110,6 +130,14 @@ def getUUID(itemName)
   click_on item
 end
 
+Quand ("l'utilisateur clique sur le bouton de création d'item à coté du corpus {string} pour creer un item {string}") do |corpus, name|
+  find_button(:id => corpus).click
+  expect(page).to have_content("undefined")
+  find_button('Ajouter un attribut').click
```
Comment (benel) [inline features/step_definitions/portfolio.rb]: Be careful the user interface has changed. Please modify the test accordingly.

File: features/step_definitions/portfolio.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
 require 'capybara/cucumber'
-require 'capybara/cuprite'
+require 'selenium/webdriver'
```
Comment (benel) [inline features/step_definitions/portfolio.rb]: Please don't change the development environment. It makes tests on the continuous integration server to fail.

File: features/item_creation.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#language: fr
+#Louis VAUCHEZ
+#Nicolas HUCHON
+#Jade VENAULT
+#Maxime TRINQUARD
```
Comment (benel) [inline features/item_creation.feature]: Authors credits are already in  the commit messages. There is no need to add them to the source files.

File: features/step_definitions/portfolio.rb [hunk 6]
```diff
@@ -1,9 +1,8 @@
...
 
 Capybara.run_server = false
-Capybara.default_driver = :cuprite
-Capybara.javascript_driver = :cuprite
+Capybara.default_driver = :selenium_chrome_headless
```
Comment (benel) [inline features/step_definitions/portfolio.rb]: Please don't change the development environment. It makes tests on the continuous integration server to fail.

File: features/item_creation.feature [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+Fonctionnalité: Créer un item
+
+Contexte:
+
+Soit le corpus "Vitraux - Bénel" rattaché au portfolio "vitraux"
+Soit "vitraux" le portfolio ouvert
+
+Scénario:
+Soit l'utilisateur "alice" connecté
+Quand l'utilisateur clique sur le bouton de création d'item à coté du corpus "Vitraux - Bénel" pour creer un item "Vitraux Test"
```
Comment (benel) [inline features/item_creation.feature]: In the scenario, each step should be related to data, not to the user interface.

File: features/item_creation.feature [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+Alors l'item "Vitraux Test" est affiché
```
Comment (benel) [inline features/item_creation.feature]: This sample data is not realistic.

**Review by benel — CHANGES_REQUESTED** [review]
Thank you for your contribution to Porphyry.
To be integrated, several changes in your code are needed however.
Thank you for taking into account every comments I made.

**Comment by louisVHZ** [comment]
@Jadestanzie @nhuchon @maxiloud 
Est-ce qu'un d'entre vous peut s'occuper de faire passer les tests pour cette fonctionnalité parce que j'arrive pas à lancer les tests sur ma machine ?
Quand les tests passeront je nettoierai les commits et on pourra merge cette branche dans le dépot officiel Porphyry.

PS : oubliez pas de `git pull` pour avoir les derniers changements sur la branche `itemCreation`

**Comment by nhuchon** [comment]
@louisVHZ je viens de modifier le test, tu avais mis la connexion avant d'afficher la page, j'ai donc du changer l'ordre pour que cela soit fonctionnel. Toutefois j'ai fait 2 commit car je me suis rendu compte que je doit modifier l'entete pour faire les tests sur ma machine, mais que le nouvel entête fait planter travis

**Comment by louisVHZ** [comment]
@nhuchon ah d'accord je comprenais pas, merci !

**Comment by nhuchon** [comment]
Attention @louisVHZ, j'ai remodifier le test car tu avais inclut une erreur lors du merge, il faudrait maintenant clean les commits

**Comment by louisVHZ** [comment]
@nhuchon Est-ce que tu peux vérifier sur ta machine en faisant un "git pull" que les tests passent bien ?

PS: Je viens de clean les commits

**Comment by nhuchon** [comment]
je fais ça dès que je le peux, je suis en train de faire la pull request de l'algèbre booleen, mais j'aurais besoin du mail associé a ton compte pour te mettre en co-auteur

**Comment by nhuchon** [comment]
@louisVHZ L'ensemble des tests sont fonctionnels chez moi

**Comment by louisVHZ** [comment]
@benel On vient de terminer les changements sur cette fonctionnalité. Il ne reste plus que votre validation pour merger la pull request.

**Pull request merged**